pycaw/tools/app_mixer/launch.py

```diff
@@ -0,0 +1,88 @@
+# ____ COM WITH MULTITHREADED APARTMENT ____
+import sys
+
+sys.coinit_flags = 0  # noqa: E402
+
+# _____ ADVANCED WINDOWS COMMUNICATION _____
+from comtypes import COMError
+
+# _________________ PYCAW _________________
+from pycaw.api.audiopolicy import IAudioSessionControl2
+
+# ______________ GUI TKINTER ______________
+from pycaw.tools.app_mixer.app_controll_and_gui import AppRow, RootFrame
+
+# ________________Register_______________
+from pycaw.tools.app_mixer.callbacks_and_manager import SessionManager
+from pycaw.utils import AudioSession, AudioUtilities
+
+
+class StartUp:
+    """
+    Register new session callback.
+    Scan for running session and add them to session_manager
+    """
+    @classmethod
+    def __init__(cls, RootFrameInstance, AppRowClass):
+
+        try:
+            mgr = AudioUtilities.GetAudioSessionManager()
+        except COMError:
+            exit("No speaker connected")
+
+        session_manager = SessionManager(RootFrameInstance, AppRowClass)
+        mgr.RegisterSessionNotification(session_manager)
+
+        # has to get called !!!
+        # to make IAudioSessionNotification::OnSessionCreated working
+        sessionEnumerator = mgr.GetSessionEnumerator()
+
+        # get all active sessions
+        count = sessionEnumerator.GetCount()
+        all_sessions = [
+            cls._get_sessions(i, sessionEnumerator) for i in range(count)
+        ]
+
+        # add sessions to session_manager
+        [session_manager.add_session(session) for session in all_sessions]
+
+        cls.mgr = mgr
+        cls.session_manager = session_manager
+
+    @classmethod
+    def unregister_all_notifications(cls):
+        print(":: cleaning up")
+        cls.mgr.UnregisterSessionNotification(cls.session_manager)
+        print(":: :: unregister SessionNotification")
+        cls.session_manager.unregister_all()
+
+    @staticmethod
+    def _get_sessions(i, sessionEnumerator):
+        # IAudioSessionControl
+        ctl = sessionEnumerator.GetSession(i)
+        ctl2 = ctl.QueryInterface(IAudioSessionControl2)
+        audio_session = AudioSession(ctl2)
+        return audio_session
+
+
+def main():
+    # ___________ CREATE ROOT WINDOW ___________
+    RootFrameInstance = RootFrame()
+
+    # ________ GET ALL ACTIVE SESSIONS ________
+    StartUp(RootFrameInstance, AppRow)
+
+    # ______________ START WINDOW ______________
+    try:
+        RootFrameInstance.mainloop()
+    except KeyboardInterrupt:
+        pass
+
+    # ________________ CLEAN UP ________________
+    StartUp.unregister_all_notifications()
+
+    print("\nTschüss")
```

Comment:
:thinking: :sweat_smile:

Reply:
yeah it doesnt seem to make much sense 😂

I will think of another class name.